docs/overview/deep_dive/formatting.rst

```diff
@@ -2,128 +2,185 @@ Formatting
-* assertion test is a :code:`tuple`, which is always :code:`true`
-* use of :code:`==` or :code:`!=` to compare :code:`str`, :code:`bytes` or :code:`int` literals
-* :code:`raise NotImplemented` should be :code:`raise NotImplementedError`
+This will install the git hook and will run the checks before you commit your code. If you want to run the checks
+manually, you can run the following command:
 
-Pre-Commit Hook
----------------
+.. code-block:: bash
 
-In Ivy, we try our best to avoid committing code with lint errors.
-To achieve this, we make use of the pre-commit package.
-The installation is explained in the `pre-commit guide`_.
+    pre-commit run --all-files
 
-The pre-commit hook runs the :code:`flake8` lint checks before each commit.
-This is efficient and useful in preventing errors being pushed to the repository.
+This will run all the required checks and will show you the output of each check.
 
-In the case where errors are found, error messages will be raised and committing will be unsuccessful until the mistake is corrected.
-If the errors are related to argument formatting, it will be reformatted automatically.
+Also when you make a commit, pre-commit will run the required checks and will show you the output of each check. If
+there are any errors, it will not allow you to commit your code. You can fix the errors and commit again.
 
-For example, the line length limit might be exceeded if arguments are all added in a single line of code like so:
+You should expect to see something similar to the following output when you run the checks:
 
-.. code-block:: python
+.. code-block:: text
 
-    def argwhere(
-        x: Union[ivy.Array, ivy.NativeArray], *, out: Optional[ivy.Array] = None,
-    ) -> ivy.Array:
+    [INFO] Stashing unstaged files to ~/.cache/pre-commit/patch1687898304-8072.
+    black....................................................................Passed
+    autoflake................................................................Passed
+    flake8...................................................................Passed
+    docformatter.............................................................Passed
+    pydocstyle...............................................................Passed
+    [INFO] Restored changes from ~/.cache/pre-commit/patch1687898304-8072.
+    [formatting-docs 3516aed563] Test commit
+    1 file changed, 1 insertion(+)
 
-When a commit is attempted, `pre-commit` would detect this error by running the lint check, and it would then reformat the arguments automatically.
+If something goes wrong, you will see the following output:
 
-.. code-block:: none
+.. code-block:: text
 
+    [INFO] Stashing unstaged files to ~/.cache/pre-commit/patch1687898304-8072.
     black....................................................................Failed
     - hook id: black
     - files were modified by this hook
 
-    reformatted ivy/functional/ivy/general.py
+    reformatted ivy\stateful\activations.py
```

Comment:
I think it might be a bit inconsistent to use the `/` separator in other parts of this file and `\` on this line

Reply:
Yup you are right, I was converting Windows format to Linux manually so I forgot about those 😅